### ISSUE TITLE
Follow-up Questions on Model Generalization

### ISSUE THREAD

**Issue by xml12138** [ArcInstitute/state#150]
HVG Selection Dependency​
I tested two HVG selection strategies on the Tahoe-100M dataset (C32 cells, 160k cells total):

- ​​(A)​​ Using the model’s predefined 2000 HVGs (directly provided)
- ​​(B)​​ Selecting new HVGs (2000 genes) from my data via sc.pp.highly_variable_genes(adata, n_top_genes=2000, flavor="seurat_v3") after normalization.

​Result:​​ Predictions **using ​strategy (B)​​** performed significantly worse than (A) in cell-eval.

<img width="969" height="217" alt="Image" src="https://github.com/user-attachments/assets/7b9094ba-6d3b-4775-8b4e-e356051faa9c" />

<img width="672" height="219" alt="Image" src="https://github.com/user-attachments/assets/fb879083-daff-44f6-8513-490c6d2071e8" />

- Does this imply that for new datasets, I must **​strictly use the model’s original 2000 HVGs​** (not dataset-specific HVGs) for preprocessing?

​Cross-Species Generalization​
The paper reports that the ​ST+HVG model was trained exclusively on human data.
 Would the model maintain reliable performance when applied to **​mouse data**​ (or other species)?

I’d appreciate insights into whether these limitations are expected or if additional steps (e.g., HVG alignment) could improve robustness.

**Comment by abhinadduri** [comment]
Hi, thanks for making an issue!

if you are using the pretrained model for inference, then it was trained specifically to output the genes in var_dims.pkl. you would have to retrain the model to model a different gene set.

we did not benchmark the model on mouse data, but it would be interesting to see how it does. there is some mouse data in the cellxgene cells used in pretraining the state embedding (SE) model

**Comment by xml12138** [comment]
Questions Regarding the 2000 Highly Variable Genes (HVGs) in ST-Parse Model:​​
**​HVG Source Clarification​**
Could you confirm whether the 2000 HVGs stored in var_dims.pkl were derived from:

1. The complete Parse dataset (all 9.69 million cells), or
2. A specific subset of the training data used for model training?

**​Practical Application Concern​**
When applying the ST-Parse model to new datasets:

1. Is it technically valid to retain these predefined 2000 HVGs even when:

- Some of these genes may not be expressed in the new data
- Certain HVGs might not be detectable through sequencing in the new samples

2.  What potential limitations or biases might arise from using fixed HVGs across different datasets?


The HVG+ST approach may have certain limitations. We are very much looking forward to the release of the model parameters and usage methods for the SE+ST model.
I appreciate your insights on maintaining model performance while addressing these practical challenges.
